### PR TITLE
added doc for CA name constraints

### PR DESCRIPTION
**Pull request by tanujd11**
CC: @maelvls

### CODE REVIEW

File: content/docs/usage/certificate.md [hunk 1]
```diff
@@ -256,6 +256,44 @@ data:
...
+  - example1.com
+  nameConstraints:
+    critical: true
+    permitted:
+      dnsDomains: ["example1.com", "example2.com"]
+      ipRanges: ["10.10.0.0/16"]
+      emailAddress: ["example@example.org"]
+    excluded:
+      ipRanges: ["10.10.0.0/24"]
+```
```
Comment (maelvls) [inline content/docs/usage/certificate.md:295]: I suggest adding a note on the fact that the self-signed issuers and CA issuers aren't validating the name constraints:

<div className="warning">

When used with cert-manager's built-in `CA` and `SelfSigned` `Issuer`, the SANs (DNS name, IP address, URI, and email address) are not checked with the certificate's own name constraints, and are not checked with any of name constraints contained in the chain of certificates the certificate belongs to.

The certificate may get issued successfully, but be rejected by clients during TLS handshakes.

</div>

File: content/docs/usage/certificate.md [hunk 2]
```diff
@@ -256,6 +256,44 @@ data:
...
+
+Root or Intermediate CA certificates can have NameConstraints. Name constraints indicates a name space within which all subject names in subsequent certificates in a certification path MUST be located.
+Checkout https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.10 for more details on this.
+
+This feature is only enabled by adding it to the
+`--feature-gates` flag on the cert-manager controller and webhook components:
+
+```bash
+--feature-gates=useCertificateRequestNameConstraints=true
+```
```
Comment (maelvls) [inline content/docs/usage/certificate.md:269]: **Suggestion:** warn the user that this feature is an alpha feature.

~~~suggestion
<div className="warning">

⛔️ The name constraints feature currently in an alpha state, and is subject to breaking changes or complete removal in future releases. This feature is only enabled by adding it to the `--feature-gates` flag on the cert-manager controller and webhook components:

```bash
--feature-gates= useCertificateRequestNameConstraints=true
```

</div>
~~~

**Review by maelvls — APPROVED** [review]
Thank you Tanuj, excellent!

/lgtm
/approve

**Comment by tanujd11** [comment]
Thanks @maelvls , can we now merge this and the cert-manager PR?

**Pull request merged**